full-service/src/service/memo.rs

```diff
@@ -0,0 +1,120 @@
+use crate::{
+    db::{
+        account::{AccountID, AccountModel},
+        models::Account,
+        WalletDbError,
+    },
+    service::ledger::{LedgerService, LedgerServiceError},
+    util::b58::{b58_decode_public_address, B58Error},
+    WalletService,
+};
+use displaydoc::Display;
+use mc_account_keys::AccountKey;
+use mc_connection::{BlockchainConnection, UserTxConnection};
+use mc_crypto_keys::RistrettoPublic;
+use mc_fog_report_validation::FogPubkeyResolver;
+use mc_transaction_extra::{MemoType, UnusedMemo};
+use std::{convert::TryFrom, ops::DerefMut};
+
+#[derive(Display, Debug)]
+#[allow(clippy::large_enum_variant, clippy::result_large_err)]
+pub enum MemoServiceError {
+    /// Error interacting with the database: {0}
+    Database(WalletDbError),
+
+    /// B58 Error
+    B58(B58Error),
+
+    /// Error decoding account key: {0}
+    Decode(mc_util_serial::DecodeError),
+
+    /// Error interacting with ledger database: {0}
+    Ledger(LedgerServiceError),
+
+    /// Key Error
+    Key(mc_crypto_keys::KeyError),
```

Comment:
Nit: This should be consistent with how we typically name and format the error variants

```suggestion
    /// WalletDb: {0}
    WalletDb(WalletDbError),

    /// B58: {0}
    B58(B58Error),

    /// Decode: {0}
    Decode(mc_util_serial::DecodeError),

    /// Ledger: {0}
    Ledger(LedgerServiceError),

    /// Key: {0}
    Key(mc_crypto_keys::KeyError),
```